dockertest/docker_daemon.py

```diff
@@ -112,51 +111,136 @@ def version(self):
 # Group of utils for managing docker daemon service.
 
 
-def start(docker_path, docker_args):
+def _which_docker():
     """
-    Start new docker daemon with special args.
+    Returns 'docker' or 'docker-latest' based on setting in
+    /etc/sysconfig/docker.
 
-    :param docker_path: Full path to executable
-    :param docker_args: List of string of command-line arguments to pass
-    :returns: Opaque daemon_process object (not for direct use)
+    Warning: this is not a reliable method. /etc/sysconfig/docker defines
+    the docker *client*; it is perfectly possible for a system to use
+    docker as client and docker-latest as daemon or vice-versa. It's
+    possible, but unsupported, so we're not going to worry about it.
     """
-    # _SpecificServiceManager creates it's methods during __init__()
-    if service.get_name_of_init() == "systemd":
-        # pylint: disable=E1101
-        utils.run("systemctl stop docker.socket", ignore_status=True)
-    service.SpecificServiceManager("docker").stop()  # pylint: disable=E1101
-    cmd = [docker_path]
-    cmd += docker_args
+    docker = 'docker'
+    with open('/etc/sysconfig/docker', 'r') as docker_sysconfig:
+        for line in docker_sysconfig:
+            if line.startswith('DOCKERBINARY='):
+                if 'docker-latest' in line:
+                    docker = 'docker-latest'
+    return docker
 
-    daemon_process = utils.AsyncJob(" ".join(cmd), close_fds=True)
-    return daemon_process
 
+def _systemd_action(action):
+    utils.run("systemctl %s %s.service" % (action, _which_docker()))
+    # FIXME: check status
 
-def output_match(daemon_process,
-                 timeout=120,
-                 regex=r"-job acceptconnections\(\) = OK \(0\)"):
+
+def stop():
+    """ stop the docker daemon """
+    _systemd_action('stop')
+
+
+def start():
+    """ start the docker daemon """
+    _systemd_action('start')
+
+
+def restart():
+    """ restart the docker daemon """
+    _systemd_action('restart')
+
+
+def _preserved_extension():
     """
-    Return True if daemon_process output matches regex within timeout period
+    File extension (including dot) used for our backup files
+    """
+    return '.docker-autotest-preserved'
```

Comment:
No, in fact I'd prefer a constant, but I have this vague recollection that Python discourages constants. If there's a best practice, I'd love to do that.